app/main/views/frameworks.py

```diff
@@ -566,3 +595,219 @@ def upload_framework_agreement(framework_slug):
         abort(503, "Framework agreement email failed to send")
 
     return redirect(url_for('.framework_agreement', framework_slug=framework_slug))
+
+
+@main.route('/frameworks/<framework_slug>/signer-details', methods=['GET', 'POST'])
+@login_required
+def signer_details(framework_slug):
+    framework = get_framework(data_api_client, framework_slug)
+    supplier_framework = return_supplier_framework_info_if_on_framework_or_abort(data_api_client, framework_slug)
+
+    form = SignerDetailsForm()
+
+    question_keys = ['signerName', 'signerRole']
+    form_errors = {}
+
+    if request.method == 'POST':
+
+        if form.validate_on_submit():
+            agreement_details = {question_key: form[question_key].data for question_key in question_keys}
+
+            data_api_client.update_supplier_framework_agreement_details(
+                current_user.supplier_id, framework_slug, agreement_details, current_user.email_address
+            )
+
+            # If they have already uploaded a file then let them go to straight to the contract review
+            # page as they are likely editing their signer details
+            if session.get('signature_page'):
+                return redirect(url_for(".contract_review", framework_slug=framework_slug))
+
+            return redirect(url_for(".signature_upload", framework_slug=framework_slug))
+        else:
+            current_app.logger.warning(
```

Comment:
I don't think we need to log this. We display form errors to the user anyway, and we generally don't log form data from unsuccessful requests.